### PR TITLE
[IMP] edi_exchange_template: Allow to use report instead of view

### PR DESCRIPTION
**Pull request by etobella**
Useful for PDFs

@simahawk

### CODE REVIEW

File: edi_exchange_template/models/edi_exchange_template_output.py [hunk 1]
```diff
@@ -18,6 +18,11 @@ class EDIExchangeOutputTemplate(models.Model):
     _inherit = "edi.exchange.template.mixin"
     _description = "EDI Exchange Output Template"
 
+    generation_type = fields.Selection(
+        [("template", "Template"), ("report", "Report")],
+        required=True,
+        default="template",
+    )
```
Comment (simahawk) [inline edi_exchange_template/models/edi_exchange_template_output.py]:
```suggestion
    generator = fields.Selection(
        [("qweb", "Qweb Template"), ("report", "Report")],
        required=True,
        default="qweb",
    )
```

File: edi_exchange_template/models/edi_exchange_template_output.py [hunk 2]
```diff
@@ -59,12 +65,20 @@ def _default_code_snippet_docs(self):
...
+            values = self._get_render_values(exchange_record, **kw)
+            output = tmpl.render(values)
+            return self._post_process_output(output)
+        if self.generation_type == "report":
+            report = self.report_id
+            values = self._get_render_values(exchange_record, **kw)
+            res_ids = values.get("res_ids", [])
+            output = report.render(res_ids, data=values)[0]
+            return self._post_process_output(output)
+        raise NotImplementedError()
```
Comment (simahawk) [inline edi_exchange_template/models/edi_exchange_template_output.py]:
```suggestion
        method = "_generate_" + self.generator
        try:
            generator = getattr(self, method)
        except AttributeError:
            raise NotImplementedError(f"`{method}` not found")
        result = generator(exchange_record, **kw)
        return self._post_process_output(result)

    def _generate_qweb(self, exchange_record, **kw):
        tmpl = self.template_id
        values = self._get_render_values(exchange_record, **kw)
        return tmpl.render(values)

    def _generate_report(self, exchange_record, **kw):
        report = self.report_id
        values = self._get_render_values(exchange_record, **kw)
        res_ids = values.get("res_ids", [])
        return report.render(res_ids, data=values)[0]
```

**Review by simahawk — CHANGES_REQUESTED** [review]
nice addition :)

**Comment by etobella** [comment]
Comments attended @simahawk

File: edi_exchange_template/views/edi_exchange_template_output_views.xml [hunk 3]
```diff
@@ -30,10 +30,22 @@
                             <field name="type_id" />
                             <field name="code" />
                             <field name="output_type" />
-                            <field name="template_id" />
+                            <field name="generator" />
+                            <field
+                                name="template_id"
+                                attrs="{'required': [('generator', '=', 'template')], 'invisible': [('generator', '!=', 'template')]}"
```
Comment (simahawk) [inline edi_exchange_template/views/edi_exchange_template_output_views.xml]:
```suggestion
                                attrs="{'required': [('generator', '=', 'qweb')], 'invisible': [('generator', '!=', 'qweb')]}"
```

**Review by simahawk — APPROVED** [review]
Small error on the view :wink: pre-approving...

**Review by albariera — APPROVED** [review]

**Pull request merged**